contracts/gauge/Gauge.sol

```diff
@@ -0,0 +1,217 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.6.7; 
+
+import "../libraries/SafeMath.sol";
+import "../libraries/SafeERC20.sol";
+import "../libraries/ProtocolGovernance.sol";
+import "../libraries/Math.sol";
+import "../libraries/reentrancy-guard.sol";
+
+
+contract Gauge is ProtocolGovernance, ReentrancyGuard {
+    using SafeMath for uint256;
+    using SafeERC20 for IERC20;
+
+    event Staked(address indexed user, uint256 amount);
+    event Withdrawn(address indexed user, uint256 amount);
+    event RewardPaid(address indexed user, uint256 reward, uint256 tokenIndex);
+    event RewardAdded(uint256 reward);
+
+    IERC20 public constant AXIAL = IERC20(0xcF8419A615c57511807236751c0AF38Db4ba3351); 
+    IERC20 public constant TREASURY = IERC20(0x4980AD7cCB304f7d3c5053Aa1131eD1EDaf48809);
+
+    IERC20 public XAXIAL = IERC20(0xcF8419A615c57511807236751c0AF38Db4ba3351); // TODO: Fix this
+    IERC20 public immutable TOKEN = IERC20(0xcF8419A615c57511807236751c0AF38Db4ba3351); // TODO: Fix this
```

Comment:
I am not sure if this will work, since staked Axial is not an ERC20.  I'll leave a related comment where this XAXIAL interface is actually used in the contract.

Reply:
Also we should update all member variables to conform to our naming convention.  If we do not yet have a convention for interfaces, I would suggest the prefix "I"